src/components/graph/network-modification-tree-model.ts

```diff
@@ -110,16 +161,16 @@ export default class NetworkModificationTreeModel {
 
         // Add children of this node recursively
         if (newNode.children) {
-            newNode.children.forEach((child) => {
-                this.addChild(child, newNode.id);
+            newNode.children.forEach((child: AbstractNode) => {
+                this.addChild(child, newNode.id, undefined, undefined);
```

Comment:
i am not sure to understand this comment, if i commit this suggestion i have the following error:
TS2554: Expected 4 arguments, but got 2

Reply:
optional parameters have to be defined like this :
optionalParameter?: string

not like this
optionalParameter: string | undefined

otherwise, you can't omit them when calling the method

Reply:
done